graph-db/serverless/src/indexCmrCollection/__test__/handler.test.js

```diff
@@ -120,53 +120,40 @@ describe('indexCmrCollection handler', () => {
     expect(statusCode).toBe(200)
   })
 
-  test('test indexing of a broken collection', async () => {
-    // Pass an error to the response of fetchCmrCollection
+  test('test collection that has an index on CMR but, it is Not stored in db', async () => {
+    const conceptId = 'C123755555-TESTPROV'
     nock(/local-cmr/)
       .get(/collections/)
-      .replyWithError('Error Collection not found', { statusCode: 404 })
```

Comment:
For the case of a 404, is that still handled?

Reply:
So this test case was intended to be if we were able to fetch the collection from CMR but, the data that it returned was invalid. I actually think this should have always been a 200 to test this functionality not a 404. We saw this happen only once against SIT but, that is why we wrapped this around a try-catch. The function fetchCmrCollection should 
` if (getConceptType(conceptId) === 'collection' && action === updateActionType) {
      try {
        const collection = await fetchCmrCollection(conceptId, token)

        const { data = {} } = collection
        const { items = [] } = data

        if (items.length === 0) {
          console.log(`Skip indexing of collection [${conceptId}] as it is not found in CMR`)

          skipCount += 1
        } else {
          const [firstCollection] = items

          // Fetch the permitted groups for this collection from access-control
          const groupList = await fetchCollectionPermittedGroups(conceptId, token)

          console.log(`Start indexing concept [${conceptId}], revision-id [${revisionId}]`)

          await indexCmrCollection(firstCollection, groupList, gremlinConnection)

          recordCount += 1
        }`
If it gets data back from CMR then it is in the else block and to test if the data in the response was not good from this function:
 https://github.com/nasa/Common-Metadata-Repository/blob/master/graph-db/serverless/src/utils/cmr/__test__/fetchCmrCollection.test.js
 
 There is a different test in that file called:
'test index of not found collection' which is should be checking against a collection which is not found from CMR in that case it would hit the `if` block. While that test does have 200 response being sent perhaps it should be 404? either way I'm going to add a more detailed comment for the test we were originally discussing. Here is the other test I mentioned: https://github.com/nasa/Common-Metadata-Repository/blob/master/graph-db/serverless/src/indexCmrCollection/__test__/handler.test.js it is called `test index of not found collection`